meinberlin/apps/dashboard2/components.py

```diff
@@ -0,0 +1,55 @@
+class DashboardComponent:
+    """Abstract interface for dashboard components.
+
+    Required properties:
+    - app_label
+
+    Required properties with default implementation:
+    - label (defaults to the lower class name)
+    - identifier (defaults to "app_label:label")
+      - has to be unique
+      - may only contain alphanumeric characters and -, _, :
+    """
+
+    app_label = None
+
+    @property
+    def label(self):
+        return self.__class__.__name__.lower()
+
+    @property
+    def identifier(self):
+        return '{app_label}:{component_label}'.format(
+            app_label=self.app_label,
+            component_label=self.label,
+        )
+
+
+class DashboardProjectComponent(DashboardComponent):
+    """Abstract interface for project components.
+
+    Inherits from DashboardComponent.
+
+    Required methods: str | None
+    - get_menu_item(project):
+      Return a localized string if the component should be rendered for the
+      project passed. Return None otherwise
+    """
+
+    def get_menu_item(self, project):
+        pass
```

Comment:
it's mainly for documentation. with the two interfaces it is always clear what to expect and what variables are present. furthermore we could check in the ProjectDispatcher if the component really is a ProjectComponent (while this could also be done with separate get_project_component and get_module_components)

but mostly i don't like method signatures like:
`def get_menu_item(self, something)`